core/rawdb/accessors_indexes.go

```diff
@@ -27,28 +29,36 @@ import (
 
 // ReadTxLookupEntry retrieves the positional metadata associated with a transaction
 // hash to allow retrieving the transaction or receipt by hash.
-func ReadTxLookupEntry(db ethdb.Reader, hash common.Hash) common.Hash {
+func ReadTxLookupEntry(db ethdb.Reader, hash common.Hash) *uint64 {
 	data, _ := db.Get(txLookupKey(hash))
 	if len(data) == 0 {
-		return common.Hash{}
+		return nil
+	}
+	//
+	if len(data) < common.HashLength {
+		var number big.Int
+		number.SetBytes(data)
+		numberU64 := number.Uint64()
+		return &numberU64
```

Comment:
Usually the canonical way to create a new big.Int is not to declare a non-pointer, rather to use `new` inline.

`number := new(big.Int).SetBytes(data).Uint64()`

Reply:
Thanks for pointing this out. Good to know!